schema/handler_test.go

```diff
@@ -126,3 +127,97 @@ func TestHandler(t *testing.T) {
 		_ = getFromTS("not-existing", http.StatusNotFound)
 	})
 }
+
+func TestHandlerAll(t *testing.T) {
+	conf, reg := internal.NewFastRegistryWithMocks(t)
+	router := x.NewRouterPublic()
+	reg.SchemaHandler().RegisterPublicRoutes(router)
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	schemas := schema.Schemas{
+		{
+			ID:     "default",
+			URL:    urlx.ParseOrPanic("file://./stub/identity.schema.json"),
+			RawURL: "file://./stub/identity.schema.json",
+		},
+		{
+			ID:     "identity2",
+			URL:    urlx.ParseOrPanic("file://./stub/identity-2.schema.json"),
+			RawURL: "file://./stub/identity-2.schema.json",
+		},
+	}
+
+	getSchemaById := func(id string) *schema.Schema {
+		s, err := schemas.GetByID(id)
+		require.NoError(t, err)
+		return s
+	}
+
+	getFromTS := func(expectCode int) []byte {
+		res, err := ts.Client().Get(fmt.Sprintf("%s/schemas", ts.URL))
+		require.NoError(t, err)
+		body, err := ioutil.ReadAll(res.Body)
+		require.NoError(t, err)
+		require.NoError(t, res.Body.Close())
+
+		require.EqualValues(t, expectCode, res.StatusCode, "%s", body)
+		return body
+	}
+
+	getFromFS := func(id string) string {
+		f, err := os.Open(strings.TrimPrefix(getSchemaById(id).RawURL, "file://"))
+		require.NoError(t, err)
+		raw, err := ioutil.ReadAll(f)
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+		return string(raw)
+	}
+
+	var schemasConfig []config.Schema
+	for _, s := range schemas {
+		if s.ID != config.DefaultIdentityTraitsSchemaID {
+			schemasConfig = append(schemasConfig, config.Schema{
+				ID:  s.ID,
+				URL: s.RawURL,
+			})
+		}
+	}
+
+	conf.MustSet(config.ViperKeyPublicBaseURL, ts.URL)
+	conf.MustSet(config.ViperKeyDefaultIdentitySchemaURL, getSchemaById(config.DefaultIdentityTraitsSchemaID).RawURL)
+	conf.MustSet(config.ViperKeyIdentitySchemas, schemasConfig)
+
+	t.Run("case=get all schemas", func(t *testing.T) {
+		body := getFromTS(http.StatusOK)
+
+		var result []struct {
+			ID     string          `json:"id"`
+			Schema json.RawMessage `json:"schema"`
+		}
+		require.NoError(t, json.Unmarshal(body, &result))
+
+		ids_orig := []string{}
+		for _, s := range schemas {
+			ids_orig = append(ids_orig, s.ID)
+		}
+		ids_list := []string{}
+		for _, s := range result {
+			ids_list = append(ids_list, s.ID)
+		}
+		for _, id := range ids_orig {
+			require.Contains(t, ids_list, id)
+		}
+
+		for _, s := range schemas {
+			for _, r := range result {
+				if r.ID == s.ID {
+					p, err := json.MarshalIndent(r.Schema, "", "  ")
+					require.NoError(t, err)
+					require.EqualValues(t, getFromFS(s.ID), string(p)+"\n")
```

Comment:
https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert#JSONEq
```suggestion
					assert.JSONEq(t, getFromFS(s.ID), string(r.Schema))
```